### PR TITLE
Misc fixes in the pivot functionallity

### PR DESCRIPTION
**Pull request by likp**
- Modified `TableApiController.cs` to:
  * Inject `IPlacementHandler` dependency.
  * Use `CommaSeparatedStringToListOfStrings` model binder for `heading` and `stub` query parameters.
  * Replace inline placement logic with `PlacementHandler`.
- Added `VariablePlacementTests` class in `VariablePlacementTests.cs` to test `PlacementHandler`.
- Introduced `PlacementHandler` class in `PlacementHandler.cs` implementing `IPlacementHandler`.
- Added `IPlacementHandler` interface in `IPlacementHandler.cs`.
- Updated `ModelStore.cs` with `CreateModelA` method for sample `PXModel`.
- Added `CommaSeparatedStringToListOfStrings` model binder in `CommaSeparatedStringToListOfStrings.cs`.
- Added `IllegalPlacementSelection` method in `ProblemUtility.cs`.
- Registered `IPlacementHandler` service in `Program.cs`.

### CODE REVIEW

File: PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            var result = new List<string>();
+
+            var keys = bindingContext.HttpContext.Request.Query.Keys.Where(x => x.Equals(modelName, StringComparison.InvariantCultureIgnoreCase));
+
+            foreach (var key in keys)
+            {
+                string? q = bindingContext.HttpContext.Request.Query[key];
+                if (q != null)
+                {
+                    var items = Regex.Split(q, ",(?=[^\\]]*(?:\\[|$))", RegexOptions.IgnoreCase,
```
Comment (JohannesFinsveen) [inline PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs:28]: Should we add try catch around the timeout add log something clever :-) 
?

File: PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs [hunk 2]
```diff
@@ -51,5 +41,16 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
...
         }
+
+        private static string CleanValue(string value)
+        {
+            var item2 = value.Trim();
+            if (item2.StartsWith('[') && item2.EndsWith(']'))
+            {
+                return value.Substring(1, item2.Length - 2);
+            }
+            return item2;
```
Comment (JohannesFinsveen) [inline PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs:52]: Should "return value.Substring(1, item2.Length - 2);"  also have a Trim() ?

Comment (likp) [reply PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs:52]: good point I will add that

File: PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs (hunk 1 above)
Comment (likp) [reply PxWeb/Code/Api2/ModelBinder/CommaSeparatedStringToListOfStrings.cs]: I think that we can let code higher up in the chain catch the error. I Changed to ReqexOptions.None instead no need for case insensitivity.

**Review by JohannesFinsveen — APPROVED** [review]
👍

**Pull request merged**